### PR TITLE
More world functions

### PR DESCRIPTION
**Pull request by theRealBatman189**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀 This PR matches 10 functions (+0.02%) on `us`.

File: src/world/area_tik/tik_15/890190.c [hunk 1]
```diff
@@ -6,8 +6,24 @@
...
-INCLUDE_ASM(s32, "world/area_tik/tik_15/890190", func_80240A78_890618);
+extern s32 D_802427BC_89235C[13][2];
 
-INCLUDE_ASM(s32, "world/area_tik/tik_15/890190", func_80240AA4_890644);
+ApiStatus func_80240A78_890618(Evt* script, s32 isInitialCall) {
+    s32* array = D_802427BC_89235C[script->varTable[0]];
 
-INCLUDE_ASM(s32, "world/area_tik/tik_15/890190", func_80240AD0_890670);
+    script->varTable[0xA] = array[0];
+    script->varTable[0xB] = array[1] ;
```
Comment (ethteck) [inline src/world/area_tik/tik_15/890190.c:15]:
```suggestion
    script->varTable[0xB] = array[1];
```

File: src/world/area_tik/tik_15/890190.c [hunk 2]
```diff
@@ -6,8 +6,24 @@
...
+
+ApiStatus func_80240AA4_890644(Evt* script) {
+    script->varTable[0] = get_item_empty_count();
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80240AD0_890670(Evt* script, s32 isInitialCall) {
+    Bytecode* args = script->ptrReadPos;
+
+    evt_set_variable(script, *args++, (gPlayerData.coins >= 64));
```
Comment (ethteck) [inline src/world/area_tik/tik_15/890190.c:27]:
```suggestion
    evt_set_variable(script, *args++, gPlayerData.coins >= 64);
```

File: src/world/area_tik/tik_18/8959F0.c [hunk 3]
```diff
@@ -2,4 +2,11 @@
 
 #include "world/common/UnkPosFunc2.inc.c"
 
-INCLUDE_ASM(s32, "world/area_tik/tik_18/8959F0", func_80240680_895D60);
+ApiStatus func_80240680_895D60(Evt* script, s32 isInitialCall) {
+    Enemy* enemy = get_enemy_safe(1);
+
+    if (enemy != 0) {
```
Comment (ethteck) [inline src/world/area_tik/tik_18/8959F0.c:8]:
```suggestion
    if (enemy != NULL) {
```

File: src/world/area_tik/tik_24/8A5A80.c [hunk 4]
```diff
@@ -1,3 +1,7 @@
 #include "tik_24.h"
 
-INCLUDE_ASM(s32, "world/area_tik/tik_24/8A5A80", func_80240000_8A5A80);
+s32 func_80240000_8A5A80(void) {
+    sprintf((s8*) &wMapShapeName, "tik_18_shape");
```
Comment (ethteck) [inline src/world/area_tik/tik_24/8A5A80.c:4]: These may be typed wrong, but the casts shouldn't be necessary. Also the rodata string for the second print I think stops after _hit and so tik_25 and tik_23 are other strings after this function. you can define them just like `const char* tik_24_str1 = "tik_25";` etc

File: src/world/area_trd/trd_04/9A88A0.c [hunk 5]
```diff
@@ -61,9 +61,49 @@
...
+    npc->collisionHeight = enemy->varTable[8] / 2;
+
+    dist2D(npc->pos.x, npc->pos.z, gPlayerStatusPtr->position.x, gPlayerStatusPtr->position.z);
+    npc->moveSpeed = 7.0f;
+    enemy->unk_10.x = npc->pos.x;
+    enemy->unk_10.y = npc->pos.y;
+    enemy->unk_10.z = npc->pos.z;
+    enemy->unk_07 = 1;
+
+    ai_enemy_play_sound(npc, 0x2C1, 0);
```
Comment (ethteck) [inline src/world/area_trd/trd_04/9A88A0.c:82]:
```suggestion
    ai_enemy_play_sound(npc, some sound ID in enums.h (please add an entry for 0x2C1 if there isn't one, 0);
```

File: src/world/area_trd/trd_04/9A88A0.c [hunk 6]
```diff
@@ -61,9 +61,49 @@
...
+    yawTemp = atan2(npc->pos.x, npc->pos.z, gPlayerStatusPtr->position.x, gPlayerStatusPtr->position.z);
+    npc->duration = 0xC;
```
Comment (ethteck) [inline src/world/area_trd/trd_04/9A88A0.c:84]:
```suggestion
    npc->duration = 12;
```

File: src/world/area_trd/trd_04/9A88A0.c [hunk 7]
```diff
@@ -61,9 +61,49 @@
...
+    npc->yaw = yawTemp;
+    script->functionTemp[0] = 0xD;
```
Comment (ethteck) [inline src/world/area_trd/trd_04/9A88A0.c:86]:
```suggestion
    script->functionTemp[0] = 13;
```

File: src/world/area_trd/trd_04/9A88A0.c [hunk 8]
```diff
@@ -61,9 +61,49 @@
...
+    npc_move_heading(npc, npc->moveSpeed, npc->yaw);
+    npc->duration--;
+
+    if (npc->duration <= 0) {
+        enemy->unk_07 = 0;
+        npc->jumpScale = 0.3f;
+        npc->jumpVelocity = 0.0f;
+        npc->moveSpeed = 3.0f;
+        npc->currentAnim.w = enemy->animList[10];
+        script->functionTemp[0] = 0xE;
```
Comment (ethteck) [inline src/world/area_trd/trd_04/9A88A0.c:104]:
```suggestion
        script->functionTemp[0] = 14;
```

**Review by ethteck — CHANGES_REQUESTED** [review]
Good stuff! More style comments

File: src/world/area_tik/tik_24/8A5A80.c (hunk 4 above)
Comment (nanaian) [reply src/world/area_tik/tik_24/8A5A80.c:4]: I have a feeling the & isnt required

File: src/world/area_trd/trd_04/9A88A0.c (hunk 5 above)
Comment (nanaian) [reply src/world/area_trd/trd_04/9A88A0.c:82]: SOUND_UNUSED_2C1...lol

**Review by nanaian — COMMENTED** [review]
Nothing to add

**Review by nanaian — APPROVED** [review]

File: src/world/area_tik/tik_24/8A5A80.c (hunk 4 above)
Comment (nanaian) [reply src/world/area_tik/tik_24/8A5A80.c:4]: Looks like `wMapShapeName` is typed wrong (`s32`) in variables.h

**Pull request merged**